clients/src/main/java/org/apache/kafka/clients/producer/Producer.java

```diff
@@ -53,6 +57,15 @@
     void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offsets,
                                   String consumerGroupId) throws ProducerFencedException;
 
+    /**
+     * See {@link KafkaProducer#sendOffsetsToTransaction(Map, ConsumerGroupMetadata)}
+     */
+    void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offsets,
+                                  ConsumerGroupMetadata groupMetadata) throws ProducerFencedException,
```

Comment:
Why we still have other exceptions declared while in `KafkaProducer` only ProducerFenced is declared?

Reply:
Will cleanup